packages-exp/auth-exp/src/core/auth/auth_impl.ts

```diff
@@ -147,18 +152,40 @@ export class AuthImpl implements Auth, _FirebaseService {
 
     // Update current Auth state. Either a new login or logout.
     await this._updateCurrentUser(user);
-    // Notify external Auth changes of Auth change event.
-    this.notifyAuthListeners();
   }
 
-  private async initializeCurrentUser(): Promise<void> {
-    const storedUser = (await this.assertedPersistence.getCurrentUser()) as User | null;
+  private async initializeCurrentUser(
+    popupRedirectResolver?: externs.PopupRedirectResolver
+  ): Promise<void> {
+    // First check to see if we have a pending redirect event.
+    let storedUser = (await this.assertedPersistence.getCurrentUser()) as User | null;
+    if (popupRedirectResolver) {
+      await this.getOrInitRedirectPersistenceManager();
```

Comment:
should we have this method return the redirectUser?

Reply:
no; the purpose of this method is to determine which user should be set as `currentUser`. The method first determines that, then updates it.